File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    count = 0
    stack = []
    
    for y in moves:
        y -= 1
        for x in range(len(board)):
            v = board[x][y]
            if v != 0:
                if stack and (stack[-1] == v):
                    stack.pop()
                    count += 2
                else:
                    stack.append(v)       
                board[x][y] = 0
                break
    
    return count

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.Stack;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int n = board.length;
        int count = 0;
        Stack<Integer> stack = new Stack<>();
        
        for (int m : moves) {
            for (int i = 0; i < n; i++) {
                int k = board[i][m-1];
                if (k == 0) continue;
                if (!stack.isEmpty() && k == stack.peek()) {
                    count += 1;
                    while (!stack.isEmpty() && k == stack.peek()) {
                        stack.pop();
                        count += 1;
                    }
                } else {
                    stack.push(k);
                }
                board[i][m-1] = 0;
                break;
            }
        }
        
        return count;
    }
}